src/base/utils/tar.cpp

```diff
@@ -0,0 +1,298 @@
+                m_end = true;
+                return TarEntry::INVALID;
+            }
+
+            if (!m_entry.isValid()) {
+                m_errorString = QCoreApplication::translate
+                                ("Utils::Tar",
+                                 "The tar file has an invalid header at position %1",
+                                 "'tar' should be left untranslated. It is a filetype.")
+                                .arg(m_pos);
+                return TarEntry::INVALID;
+            }
+
+            if (!m_entry.isVerified()) {
+                m_errorString = QCoreApplication::translate
+                                ("Utils::Tar",
+                                 "The tar file has a header with unverified checksum at position %1",
+                                 "'tar' should be left untranslated. It is a filetype.")
+                                .arg(m_pos);
+                return TarEntry::INVALID;
+            }
+
+            const int fileSize = m_entry.size();
+            const int availableSize = m_data.size() - m_pos - TAR_HEADER_SIZE;
+            if (availableSize < fileSize) {
+                m_errorString = QCoreApplication::translate
+                                ("Utils::Tar",
+                                 "The tar file has ended prematurely",
+                                 "'tar' should be left untranslated. It is a filetype.");
+                return TarEntry::INVALID;
+            }
+
+            return m_entry.type();
+        }
+
+        QString name() const
+        {
+            if (atEnd() || hasError() || m_entry.isNull())
+                return {};
+
+            return m_entry.name();
+        }
+
+        QByteArray fileData() const
+        {
+            if (atEnd() || hasError() || m_entry.isNull())
+                return {};
+
+            const int fileSize =  m_entry.size();
+            const int filePos = m_pos + TAR_HEADER_SIZE;
+            return {m_data.constData() + filePos, fileSize};
+        }
+
+        TarEntry::Types type() const
+        {
+            if (atEnd() || hasError() || m_entry.isNull())
+                return TarEntry::INVALID;
+
+            return m_entry.type();
+        }
+
+        bool atEnd() const { return m_end; }
+        bool hasError() const { return !errorString().isEmpty(); }
+        QString errorString() const { return m_errorString; }
+
+    private:
+        bool m_end = false;
+        int m_pos = 0;
+        const QByteArray m_data;
+        QString m_errorString;
+        TarEntry m_entry;
+    };
+}
+
+QHash<QString, QByteArray> Utils::Tar::untar(const QByteArray &data, QString &errorString)
+{
+    QHash<QString, QByteArray> files;
+    TarStreamReader reader(data);
+    while (reader.readNext() != TarEntry::INVALID) {
+        if (!(reader.type() == TarEntry::REG_FILE_ASCII || reader.type() == TarEntry::REG_FILE_NULL))
+            continue;
+
+        files.insert(reader.name(), reader.fileData());
```

Comment:
I know this was discussed before. However I would still like the returned values to be QByteArray views.
Adding a comment about this function is returning a view (that the input `data` is required to be alive in client code in order for the returned value to be meaningful) would suffice for me. And then the return type will be expected to become `QHash<QString, const QByteArray>`